docs/install/Knative-with-any-k8s.md

```diff
@@ -1,89 +1,470 @@
+
+The following commands install Gloo and enables its Knative integration:
+
+1. Make sure `glooctl` is installed (version 1.3.x and higher recommended):
 
    ```bash
-   kubectl apply --filename https://github.com/knative/serving/releases/download/{{< version >}}/serving.yaml \
-   --filename https://github.com/knative/eventing/releases/download/{{< version >}}/eventing.yaml \
-   --filename https://github.com/knative/serving/releases/download/{{< version >}}/monitoring.yaml
+   glooctl version
    ```
 
+   If it is not installed, follow the instructions [here](https://docs.solo.io/gloo/latest/installation/knative/#install-command-line-tool-cli)
+
+2. Install Gloo and the Knative integration:
+
+    ```bash
+    glooctl install knative --install-knative=false
+    ```
+
+3. Fetch the External IP or CNAME
+
+   ```bash
+   kubectl get svc -ngloo-system knative-external-proxy
+   ```
+
+   Save this for configuring DNS below.
+
+{{< /tab >}}
+
+{{% tab name="Istio" %}}
+The following commands install Istio and enables its Knative integration:
+
+1. [Installing Istio for Knative](./installing-istio.md)
+
+2. Install the Knative Istio controller:
+
+   ```bash
+   kubectl apply --filename https://github.com/knative/serving/releases/download/{{< version >}}/serving-istio.yaml
+   ```
+
+3. Fetch the External IP or CNAME
+
+   ```bash
+   kubectl --namespace istio-system get service istio-ingressgateway
+   ```
+
+   Save this for configuring DNS below.
+
+{{< /tab >}}
+
+{{% tab name="Kourier" %}}
+The following commands install Kourier and enables its Knative integration:
+
+1. Install the Knative Kourier controller:
+
+   ```bash
+   kubectl apply --filename https://raw.githubusercontent.com/knative/serving/{{< version >}}/third_party/kourier-latest/kourier.yaml
+   ```
+
+2. To configure Knative Serving to use Kourier by default:
+
+    ```bash
+    kubectl patch configmap/config-network \
+      --namespace knative-serving \
+      --type merge \
+      --patch '{"data":{"ingress.class":"kourier.ingress.networking.knative.dev"}}'
+    ```
+
+3. Fetch the External IP or CNAME
+
+   ```bash
+   kubectl --namespace kourier-system get service kourier
+   ```
+
+   Save this for configuring DNS below.
+
+{{< /tab >}}
+{{< /tabs >}}
+
+1. Configure DNS
+
+   <!-- This indentation is important for things to render properly. -->
+   {{< tabs name="serving_dns" >}}
+{{% tab name="Magic DNS (xip.io)" %}}
+We ship a simple Kubernetes Job called "default domain" that will (see caveats) configure Knative Serving to use <a href="http://xip.io">xip.io</a> as the default DNS suffix.
+
+```bash
+kubectl apply --filename https://github.com/knative/serving/releases/download/{{< version >}}/serving-default-domain.yaml
+```
+
+**Caveat**: This will only work if the cluster LoadBalancer service exposes an IPv4 address, so it will not work with IPv6 clusters, AWS, or local setups like Minikube.  For these, see "Using a real DNS record" (below).
```

Comment:
```suggestion
**Caveat**: This will only work if the cluster LoadBalancer service exposes an IPv4 address, so it will not work with IPv6 clusters, AWS, or local setups like Minikube.  For these, see "Using a real DNS record".
```

I don't think this is (below) when it is in tab form?

Reply:
Good catch :D